src/network/server_tests.rs

```diff
@@ -276,5 +276,58 @@ mod tests {
             .await
             .unwrap_err();
         assert_eq!(response.code(), tonic::Code::NotFound);
+
+        // Returns all active casts
+        let all_casts_request = proto::FidRequest {
+            fid: SHARD1_FID,
+            page_size: None,
+            page_token: None,
+            reverse: None,
+        };
+        let response = service
+            .get_casts_by_fid(Request::new(all_casts_request))
+            .await;
+        test_helper::assert_contains_all_messages(&response, &[&cast_add2]);
+
+        // Pagination works
+        let all_casts_request = proto::FidRequest {
+            fid: SHARD1_FID,
+            page_size: Some(1),
+            page_token: None,
+            reverse: None,
+        };
+        let response = service
+            .get_casts_by_fid(Request::new(all_casts_request))
+            .await;
+        test_helper::assert_contains_all_messages(&response, &[&cast_add2]);
+
+        // TODO: Fix pagination
```

Comment:
I'm not entirely sure why but I think the rocksdb pagination is broken. Will investigate tomorrow. cc @aditiharini